src/block/decompress.rs

```diff
@@ -286,7 +262,7 @@ fn decompress_internal<const USE_DICT: bool>(
             let mut start_ptr = unsafe { output_ptr.sub(offset) };
             #[cfg(feature = "checked-decode")]
             {
-                if unsafe { start_ptr.add(ext_dict.len()) } < output_base {
+                if offset == 0 || unsafe { start_ptr.add(ext_dict.len()) } < output_base {
```

Comment:
We are missing fuzzy tests with the dictionary.
I think this didn't cause out of bounds reads, can you confirm?

Reply:
The fuzz_decomp_corrupt_frame sort of covered it indirectly (but not much) through but I'll push a small improvement shortly.

Offset 0 wouldn't cause an OOB access but could reveal uninitiated data in the output buffer. Probably only applicable for Vec + Vec::set_len outputs used previously.